File: src/server/engine/core/domain/src/entities/item.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Coordinates {
    pub lat: f64,
    pub lng: f64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Item {
    pub id: String,
    pub title: String,
    pub description: Option<String>,
    pub status: String,
    pub tags: Vec<String>,
    pub coordinates: Option<Coordinates>,
    pub rating: Option<f64>,
    pub comment_count: Option<i64>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum ItemAction {
    Create,
    Update,
    Delete,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ItemEvent {
    pub action: ItemAction,
    pub item: Item,
}

File: src/client/sdk/core/src/entities/item.ts
export interface Coordinates {
  lat: number;
  lng: number;
}

export interface Item {
  id: string;
  title: string;
  description: string | null;
  status: string;
  tags: string[];
  coordinates: Coordinates | null;
  rating: number | null;
  comment_count: number | null;
  created_at: string;
  updated_at: string;
}
